lib/mix/tasks/compile/live_view.ex

```diff
@@ -0,0 +1,81 @@
+defmodule Mix.Tasks.Compile.LiveView do
+  use Mix.Task
+  @recursive true
+
+  alias Mix.Task.Compiler.Diagnostic
+
+  @doc false
+  def run(args) do
+    {compile_opts, _argv, _errors} = OptionParser.parse(args, strict: [return_errors: :boolean])
+
+    diagnostics = validate_components_calls(project_modules())
+
+    if !compile_opts[:return_errors] do
+      print_diagnostics(diagnostics)
+    end
+
+    {:ok, diagnostics}
```

Comment:
Shouldn't we return `:error` if diagnostics is non empty?

Reply:
From the editor's PoV, as far as I can remember, it expects `{:ok, diagnostics}` because it means the compiler run as expected, collecting errors/warnings from the code. While with `{:error, ...}`, it would mean that something went wrong with the compiler itself. I might be wrong though. If that's the case, I'll probably need to change how ElixirLS handles that.

Reply:
I am pretty sure `{:error, ...}` means the code being compiled fail and ElixirLS should already handle it because that's what the Elixir compiler returns on errors.